packages/patternfly-4/react-core/src/components/Form/FormGroup.js

```diff
@@ -0,0 +1,134 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import styles from '@patternfly/patternfly-next/components/Form/form.css';
+import { ASTERISK } from '../../internal/htmlConstants';
+import { FormContext } from '../Form/FormContext';
+import { css, getModifier } from '@patternfly/react-styles';
+
+const childrenArrayContains = (childrenArray, attribute, errorMessage) => {
+  let throwError = true;
+  childrenArray.forEach(child => {
+    if (child.props[attribute]) {
+      throwError = false;
+    }
+  });
+  if (throwError) {
+    return new Error(errorMessage);
+  }
+  return null;
+};
+
+const propTypes = {
+  /** Anything that can be rendered as FormGroup content. */
+  children: props => {
+    if (typeof props.label !== 'string') {
+      if (Array.isArray(props.children)) {
+        return childrenArrayContains(
+          props.children,
+          'aria-label',
+          `When label is not a string, at least one of Form Group children should have an aria-label attribute.`
+        );
+      } else if (React.isValidElement(props.label)) {
+        if (!props.children.props['aria-label']) {
+          return new Error(`When label is not a string, a Form Group child should have an aria-label attribute.`);
+        }
+      }
+    } else if (Array.isArray(props.children)) {
+      return childrenArrayContains(props.children, 'id', `At least one of Form Group children should have an id.`);
+    } else if (!props.children.props.id) {
+      return new Error(`A child element of Form Group has to have an id.`);
+    }
+    return null;
+  },
+  /** Additional classes added to the FormGroup. */
+  className: PropTypes.string,
+  /** Label text before the field. */
+  label: PropTypes.oneOfType([PropTypes.string, PropTypes.node]),
+  /** Sets the FormGroup required. */
+  isRequired: PropTypes.bool,
+  /** Sets the FormGroup isValid. */
+  isValid: PropTypes.bool,
+  /** Sets the FormGroup isInline. */
+  isInline: PropTypes.bool,
+  /** Helper text after the field. It can be a simple text, a function or an object. */
+  helperText: PropTypes.oneOfType([PropTypes.string, PropTypes.node]),
+  /** Helper text after the field when the field is isValid. It can be a simple text, a function or an object. */
+  helperTextInvalid: PropTypes.oneOfType([PropTypes.string, PropTypes.node]),
+  /** ID of the included field. It has to be the same for proper working. */
+  fieldId: PropTypes.string.isRequired
+};
+
+const defaultProps = {
+  children: null,
+  className: '',
+  label: undefined,
+  isRequired: false,
+  isValid: true,
+  isInline: false,
+  helperText: undefined,
+  helperTextInvalid: undefined,
+  fieldId: undefined
+};
+
+const renderHelperText = ({ helperText, classes, fieldId }) => {
+  if (typeof helperText === 'object') {
+    return helperText;
```

Comment:
Same here. Elements should be inside the helper text block so they are styled. Since this will remove the conditional it would be better to simply inline in the main render.